test/Microsoft.ML.Tests/Transformers/PcaTests.cs

```diff
@@ -62,7 +62,7 @@ public void TestPcaEstimator()
             using (var ch = _env.Start("save"))
             {
                 IDataView savedData = TakeFilter.Create(_env, est.Fit(data.AsDynamic).Transform(data.AsDynamic), 4);
-                savedData = SelectColumnsTransform.CreateKeep(_env, savedData, "pca");
+                savedData = SelectColumnsTransform.CreateKeep(_env, savedData, new[] { "pca" });
```

Comment:
>new[] [](start = 79, length = 5)

is it unavoidable?

Reply:
If we want the "keepHidden" argument to be optional, then the string[] argument can't be a 'params' argument. I can bring back the other CreateKeep overload with the non-optional keepHidden argument if you think it's important.

---
In reply to: [231233064](https://github.com/dotnet/machinelearning/pull/1505#discussion_r231233064) [](ancestors = 231233064)

Reply:
I thought that you can have optional and then `params`

---
In reply to: [231242980](https://github.com/dotnet/machinelearning/pull/1505#discussion_r231242980) [](ancestors = 231242980,231233064)

Reply:
..these days

---
In reply to: [231257225](https://github.com/dotnet/machinelearning/pull/1505#discussion_r231257225) [](ancestors = 231257225,231242980,231233064)

Reply:
Turns out you're right, it is possible to have an optional followed by params. But to use it without specifying the optional it would have to be: keepColumns: new[] {"a", "b", "c"}, which is longer than just adding the "new[]"...

---
In reply to: [231257259](https://github.com/dotnet/machinelearning/pull/1505#discussion_r231257259) [](ancestors = 231257259,231257225,231242980,231233064)